kv/dist_sender_server_test.go

```diff
@@ -188,3 +188,34 @@ func TestStartEqualsEndKeyScan(t *testing.T) {
 		t.Fatalf("expected error on scan with startkey == endkey")
 	}
 }
+
+// TestSplitByMeta2KeyMax check range splitting at key Meta2KeyMax should
+// failed as Meta2KeyMax is not an invalid split key.
+func TestSplitByMeta2KeyMax(t *testing.T) {
+	s := startServer(t)
+	db := createTestClient(t, s.ServingAddr())
+	db.User = storage.UserRoot
+	defer s.Stop()
+
+	ch := make(chan struct{})
+	go func() {
+		if err := db.Run(client.Call{
+			Args: &proto.AdminSplitRequest{
+				RequestHeader: proto.RequestHeader{
+					Key: proto.KeyMin,
+				},
+				SplitKey: keys.Meta2KeyMax,
+			},
+			Reply: &proto.AdminSplitResponse{},
+		}); err == nil {
+			t.Fatalf("range split on Meta2KeyMax should fail")
+		}
+		close(ch)
+	}()
+
+	select {
+	case <-ch:
+	case <-time.After(100 * time.Millisecond):
```

Comment:
removing this timeout makes the test pass for me. I think 100ms is just too aggressive.